### PR TITLE
fix: remove db password from logs

### PR DESCRIPTION
**Pull request by anku255**
## Summary of change

This PR removes the db password from being logged.

## Related issues
- https://github.com/supertokens/supertokens-core/issues/907

## Test Plan
(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes
(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates
- [x] Changelog has been updated
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [x] Changes to the version if needed
   - In `build.gradle`
- [x] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them in `implementationDependencies.json`.
- [x] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

### CODE REVIEW

File: src/main/java/io/supertokens/storage/mysql/config/MySQLConfig.java [hunk 1]
```diff
@@ -509,10 +509,11 @@ public String getConnectionPoolId() {
...
         for (Field field : MySQLConfig.class.getDeclaredFields()) {
             if (field.isAnnotationPresent(ConnectionPoolProperty.class)) {
-                connectionPoolId.append("|");
                 try {
-                    if (field.get(this) != null) {
-                        connectionPoolId.append(field.get(this).toString());
+                    String fieldName = field.getName();
+                    String fieldValue = field.get(this) != null ? field.get(this).toString() : null;
+                    if (!fieldName.equals("mysql_password") && fieldValue != null) {
+                        connectionPoolId.append("|" + fieldValue);
```
Comment (rishabhpoddar) [inline src/main/java/io/supertokens/storage/mysql/config/MySQLConfig.java:516]: i don't think this is the right thing to do cause the connection id still needs to have the password to uniquely differentiate it from other attempted connections to the same db. So instead, we should change the logging class to not log the password in any string.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: src/main/java/io/supertokens/storage/mysql/config/MySQLConfig.java (hunk 1 above)
Comment (anku255) [reply src/main/java/io/supertokens/storage/mysql/config/MySQLConfig.java]: Fixed.

**Review by rishabhpoddar — APPROVED** [review]

File: src/test/java/io/supertokens/storage/mysql/test/LoggingTest.java [hunk 2]
```diff
@@ -308,6 +310,190 @@ public void confirmHikariLoggerClosedOnlyWhenProcessEnds() throws Exception {
...
+        }
+
+        File errorLog = new File(Config.getConfig(main).getErrorLogPath(main));
+
+        boolean dbPasswordMaskedInErrorLog = false;
+
+        try (Scanner errorScanner = new Scanner(errorLog, StandardCharsets.UTF_8)) {
+            while (errorScanner.hasNextLine()) {
+                String line = errorScanner.nextLine();
+                System.out.println("line: " + line);
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/storage/mysql/test/LoggingTest.java:440]: remove this.

File: src/test/java/io/supertokens/storage/mysql/test/LoggingTest.java [hunk 3]
```diff
@@ -308,6 +310,190 @@ public void confirmHikariLoggerClosedOnlyWhenProcessEnds() throws Exception {
...
+
+        Utils.setValueInConfig("mysql_user", dbUser);
+        Utils.setValueInConfig("mysql_password", dbPassword);
+        Utils.setValueInConfig("mysql_database_name", dbName);
+
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args);
+
+        process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.INIT_FAILURE);
+
+        File errorLog = new File(Config.getConfig(process.getProcess()).getErrorLogPath(process.getProcess()));
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/storage/mysql/test/LoggingTest.java:368]: this is not the best way to observe error logs. The better way is to overwrite the logging mechanicm to write to a buffer you create in this test and then just read that. There are tests in the core that do this already like this one: https://github.com/supertokens/supertokens-core/blob/master/src/test/java/io/supertokens/test/LoggingTest.java#L217

Comment (rishabhpoddar) [reply src/test/java/io/supertokens/storage/mysql/test/LoggingTest.java:368]: This comment applies to all new tests you have added that do this.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**